pgrx/src/lib.rs

```diff
@@ -182,7 +182,13 @@ const _: () = {
 macro_rules! pg_module_magic {
     () => {
         $crate::pg_magic_func!();
-        $crate::pg_sql_graph_magic!();
+
+        // A marker function which must exist in the root of the extension for proper linking by the
+        // "pgrx_embed" binary during `cargo-pgrx schema` generation.
+        #[doc(hidden)]
+        pub fn __pgrx_marker() {
```

Comment:
this should probably be `#[inline(never)`].

I haven't been able to cause it to happen, but it seems to me that the compiler's DCE passes could somehow cause the `pgrx_embed` binary to not link to the .so if the function and its call point in `pgrx_embed::main()` were removed.

Reply:
That is very astute, as Saethlin has been working extensively on making the compiler do _very aggressive_ inlining, because idiomatic Rust uses lots of tiny functions.